### PR TITLE
Allow projects to link to multiple style databases, show all symbols and entities from those databases

### PR DESCRIPTION
**Pull request by nyalldawson**
This new feature allows users to set a list of QGIS style databases for a project. The style paths can point to style .db databases or .xml exports, located e.g. on a shared folder or alongside a project. When that project is loaded, symbol and label style selectors will show ALL symbols from ALL styles linked to the project. Additionally, every project now has a project-specific style database (which is stored alongside the project), for storing project-specific symbols, text formats, etc.

This has a few use cases:

1. A style database can be created eg for a single client, containing all the symbols used for that particular client and which are nice to share across different projects, but ONLY projects for that client (and you don't want to clutter up your default style database with those client-specific styles)
2. Multi user environments could store project specific styles in a shared location, and link a project to these styles. Then ALL users of the project will have the common symbology available to them, without having to manually import/update their local style database.
3. By allowing symbols to be stored in a project file, it avoids users cluttering up their global style database with project-specific symbology. It also ensures that other users of this project will also have all the required symbology available for immediate re-use when loading that project.

**This functionality requires a QGIS build based on Qt 5.13 or later, and is hidden on older builds (e.g. on Ubuntu 20.04)**

Sponsored by the QGIS Swiss user group.

Screencasts:

Saving a symbol into the project database:

https://user-images.githubusercontent.com/1829991/168707911-b6867ca4-8b4f-4afe-87e8-ce7903761db6.mp4

Adding additional shared style libraries to a project:

https://user-images.githubusercontent.com/1829991/168708342-085c5009-4c1c-4fbe-8e58-92dcbb3e4557.mp4

### CODE REVIEW

**Comment by nyalldawson** [comment]
Also should add: this requires a QGIS build based on Qt 5.13 or greater only

**Comment by nirvn** [comment]
My two cents: +1 on freeze exception. It'd go along very well with the default style revamp done last week and as pointed out does fill an important gap with our style handling in the context of multi user environment.

File: src/gui/symbology/qgsstylemanagerdialog.cpp [hunk 1]
```diff
@@ -213,49 +251,26 @@ QgsStyleManagerDialog::QgsStyleManagerDialog( QgsStyle *style, QWidget *parent,
...
-           );
-  }
-  else
+  connect( btnAddItem, &QPushButton::clicked, this, [ = ]( bool )
   {
-    btnAddTag->setEnabled( false );
-    btnAddSmartgroup->setEnabled( false );
+    // only show add item if the btn doesn't have a menu -- otherwise it should show the menu instead!
+    if ( !btnAddItem->menu() )
+    { addItem(); }
```
Comment (nirvn) [inline src/gui/symbology/qgsstylemanagerdialog.cpp:259]: Ew formatting :wink: we gotta tame astyle to handle this better one day :)

File: src/gui/symbology/qgsstylemanagerdialog.cpp [hunk 2]
```diff
@@ -518,18 +460,166 @@ QgsStyleManagerDialog::QgsStyleManagerDialog( QgsStyle *style, QWidget *parent,
...
+
+  QgsCheckableStyleModel *oldModel = mModel;
+
+  mStyle = style;
+  const bool readOnly = isReadOnly();
+  if ( mStyle != QgsStyle::defaultStyle() )
+  {
+    if ( !mActionCopyToDefault )
+    {
+      mActionCopyToDefault = new QAction( tr( "Copy Selection to Default Style…" ), this );
```
Comment (nirvn) [inline src/gui/symbology/qgsstylemanagerdialog.cpp:485]: Would it be useful to have a Copy Selection to Project Style when the current style isn't the project style?

**Review by nirvn — APPROVED** [review]
Reviewed, looking good.

File: src/gui/symbology/qgsstylemanagerdialog.cpp (hunk 2 above)
Comment (nyalldawson) [reply src/gui/symbology/qgsstylemanagerdialog.cpp:487]: I'll defer this to a follow up

**Pull request merged**